### PR TITLE
SAMZA-2220: Startpoints - Fully encapsulate resolution of starting offsets in OffsetManager

### PR DESCRIPTION
**Pull request by dnishimura**
In PR #1026, the startpoint resolution was integrated with only the system consumers, but not the task instances "starting offset caught up logic". Without the correct integration with the task instance caught up logic, the startpoint resolution from PR #1026 is ignored.

This PR encapsulates the startpoint resolution fully in the offset manager so that all components dependent on the starting offsets will get a consistent resolution of the offsets.

### CODE REVIEW

File: samza-core/src/main/scala/org/apache/samza/checkpoint/OffsetManager.scala [hunk 1]
```diff
@@ -193,6 +194,7 @@ class OffsetManager(
     stripResetStreams
     loadStartingOffsets
     loadStartpoints
+    resolveStartpointsToStartingOffsets
```
Comment (rmatharu) [inline samza-core/src/main/scala/org/apache/samza/checkpoint/OffsetManager.scala]: Would make sense to call resolveStartpointsToStartingOffsets  from loadStartpoints?

Comment (dnishimura) [reply samza-core/src/main/scala/org/apache/samza/checkpoint/OffsetManager.scala]: Initially had it after loading defaults. Makes better sense to put it in loadStartpoints. Good catch, thanks!

**Comment by dnishimura** [comment]
@shanthoosh - please take a look when you get a chance.

File: samza-core/src/main/scala/org/apache/samza/checkpoint/OffsetManager.scala [hunk 2]
```diff
@@ -535,9 +536,47 @@ class OffsetManager(
         startpoints
           .foreach(taskMap => taskMap._2
             .foreach(sspMap => info("Loaded startpoint: %s for SSP: %s and task: %s" format (sspMap._2, sspMap._1, taskMap._1))))
+        resolveStartpointsToStartingOffsets
```
Comment (shanthoosh) [inline samza-core/src/main/scala/org/apache/samza/checkpoint/OffsetManager.scala]: Minor: Would be better to update the java doc of the `OffsetManager` class and include that it accounts for startpoints into calculation of the starting offset.

Comment (dnishimura) [reply samza-core/src/main/scala/org/apache/samza/checkpoint/OffsetManager.scala]: Done. Thanks for the feedback.

**Comment by shanthoosh** [comment]
@dnishimura 
Thanks for the patch. Merged to trunk.

**Pull request merged**